src/test/java/org/assertj/jodatime/api/localdate/LocalDateAssert_isAfterOrEqualTo_Test.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2018 the original author or authors.
+ */
+package org.assertj.jodatime.api.localdate;
+
+import org.joda.time.LocalDate;
+import org.junit.Test;
+import org.junit.experimental.theories.Theories;
+import org.junit.experimental.theories.Theory;
+import org.junit.runner.RunWith;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+import static org.assertj.jodatime.api.Assertions.assertThat;
+
+/**
+ * @author Evgenii Strepetov
+ */
+@RunWith(Theories.class)
+public class LocalDateAssert_isAfterOrEqualTo_Test extends LocalDateAssertBaseTest {
+
+  @Theory
+  public void test_isAfterOrEqual_assertion(LocalDate referenceDate, LocalDate dateBefore, LocalDate dateAfter) {
+    // GIVEN
+    testAssumptions(referenceDate, dateBefore, dateAfter);
+    // WHEN
+    assertThat(dateAfter).isAfterOrEqualTo(referenceDate);
+    assertThat(dateAfter).isAfterOrEqualTo(referenceDate.toString());
+    assertThat(referenceDate).isAfterOrEqualTo(referenceDate);
+    assertThat(referenceDate).isAfterOrEqualTo(referenceDate.toString());
+    // THEN
+    verify_that_isAfterOrEqual_assertion_fails_and_throws_AssertionError(dateBefore, referenceDate);
+  }
+
+  @Test
+  public void test_isAfterOrEqual_assertion_error_message() {
```

Comment:
Since assertj-joda-time relies on assertj-core 3.x, we can take advantage to [`catchThrowable`](http://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html#exception-assertion), that will make the test easier to read and not having to use `failBecauseExpectedAssertionErrorWasNotThrown`.
I'm keen on using a BBD style test (GIVEN, WHEN THEN).

This comment applies to any other test checking for an expected `AssertionError`

Reply:
Refactored to assertThatThrownBy() everywhere it is applicable